src/query/sql/src/planner/optimizer/rule/rewrite/rule_fold_count_aggregate.rs

```diff
@@ -87,6 +87,17 @@ impl Rule for RuleFoldCountAggregate {
                 _ => false,
             });
 
+        let simple_nullable_count = agg.group_items.is_empty()
+            && agg.aggregate_functions.iter().all(|agg| match &agg.scalar {
+                Scalar::AggregateFunction(agg_func) => {
+                    agg_func.func_name == "count"
+                        && (agg_func.args.is_empty()
+                            || matches!(agg_func.args[0].data_type(), Nullable(_)))
+                        && !agg_func.distinct
+                }
+                _ => false,
+            });
+
         if let (true, Some(card)) = (is_simple_count, input_prop.precise_cardinality) {
```

Comment:
why not introduce another field like `    pub precise_cardinality: Option<u64>,`